gateway/server.go

```diff
@@ -902,16 +902,34 @@ func (gw *Gateway) createResponseMiddlewareChain(spec *APISpec, responseFuncs []
 		baseHandler     = BaseTykResponseHandler{Spec: spec, Gw: gw}
 	)
 	gw.responseMWAppendEnabled(&responseMWChain, &ResponseTransformMiddleware{BaseTykResponseHandler: baseHandler})
+
+	headerInjector := &HeaderInjector{BaseTykResponseHandler: baseHandler}
+	headerInjectorAdded := gw.responseMWAppendEnabled(&responseMWChain, headerInjector)
 	for _, processorDetail := range spec.ResponseProcessors {
+		// TODO: This if statement will be removed in 5.4 as header_injector response processor will be removed
```

Comment:
Don't leave TODOs in code- if you want, move the code bit out into it's own function, and add a `// Deprecated:` comment to that function (make it easy to remove).

Reply:
Removed the TODO. However, the if statement is not very proper to move to a function, there are multiple variables already so I prefer going inline instead of function.